app/assets/js/app/modules/printlistener.js

```diff
@@ -0,0 +1,22 @@
+import domready from './domready'
+
+export default function initPrintlistener() {
+  let printed = false
+  const onafterPrinting = () => {
+    if (!printed) {
+      window.dataLayer.push({ 'event': 'onAfterPrint', 'currentURL': location.pathname })
+      printed = true
+    }
+  }
+  if (window.matchMedia) {
+    let mediaQueryList = window.matchMedia('print')
+    mediaQueryList.addListener(function(mql) {
+      if (mql.matches) {
```

Comment:
Working as expected, however, the `onafterprint` with `mql` is when it is `if (!mql.matches) {`. <- this is how the current analytics event is done. 
https://github.com/ONSdigital/sdc-global-design-patterns/blob/fb4d79d15ffca96674c8c06e2245ea93ba5531a7/assets/js/analytics.js#L48-L64

```js
(function() {
    var beforePrint = function() {
        console.log('Functionality to run before printing.');
    };
    var afterPrint = function() {
        console.log('Functionality to run after printing');
    };

    if (window.matchMedia) {
        var mediaQueryList = window.matchMedia('print');
        mediaQueryList.addListener(function(mql) {
            if (mql.matches) {
                beforePrint();
            } else {
                afterPrint();
            }
        });
    }

    window.onbeforeprint = beforePrint;
    window.onafterprint = afterPrint;
}());
```
https://www.tjvantoll.com/2012/06/15/detecting-print-requests-with-javascript/

Reply:
Not a blocker but `addListener ` is also deprecated, ideally, we would use https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/onchange.

Reply:
Suggested changes now implemented